### DISCUSSION TITLE
data key inside  "await prisma.some_model.create({" object is giving typescript error

### DISCUSSION THREAD

**Discussion by vipertecpro** [prisma/prisma#20034] [Q&A] [answered]
It's working fine, but in phpstrom it's giving error which is very annoying, can someone help me out with that 

![Screenshot 2023-07-01 181240](https://github.com/prisma/prisma/assets/11619188/502535a3-59d5-4791-9638-5c07bcf4321c)

this is the code 

```
export async function POST(request: Request) {
    const body: RequestBody = await request.json();

    const user = await prisma.user.create({
        data : {
            name: body.name,
            email: body.email,
            password: await bcrypt.hash(body.password, 10)
        }
    });
    const { password, ...result } = user;
    return new Response(JSON.stringify(result));
}
```

This is the error message from the screenshot 

TS2322: Type '{ name: string; email: string; password: string; }' is not assignable to type '(Without<UserCreateInput, UserUncheckedCreateInput> & UserUncheckedCreateInput) | (Without<...> & UserCreateInput)'.   Type '{ name: string; email: string; password: string; }' is not assignable to type 'Without<UserUncheckedCreateInput, UserCreateInput> & UserCreateInput'.     Type '{ name: string; email: string; password: string; }' is missing the following properties from type 'UserCreateInput': created_at, updated_at  index.d.ts(1893, 5): The expected type comes from property 'data' which is declared here on type '{ select?: UserSelect<DefaultArgs> | null | undefined; include?: UserInclude<DefaultArgs> | null | undefined; data: (Without<UserCreateInput, UserUncheckedCreateInput> & UserUncheckedCreateInput) | (Without<...> & UserCreateInput); }'

**Comment by ludralph** [comment, marked as answer]
Hi @vipertecpro 👋 

From the error message, the provided value for data is { name: string; email: string; password: string; }, which is missing the created_at and updated_at properties required by the UserCreateInput type.

To resolve this issue, you need to include the created_at and updated_at properties in the data object. You can set their values to the current date/time or an appropriate default value.

For example, the updated code should look like
```js
export async function POST(request: Request) {
    const body: RequestBody = await request.json();

    const user = await prisma.user.create({
        data: {
            name: body.name,
            email: body.email,
            password: await bcrypt.hash(body.password, 10),
            created_at: new Date(),
            updated_at: new Date()
        }
    });
    const { password, ...result } = user;
    return new Response(JSON.stringify(result));
}
```
You may need to adjust the created_at and updated_at values as per your requirements.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by vipertecpro** [reply]
ohh i get it now i need to align all the prisma model schema and thank you so much for the quick reply i will be in touch

**Reply by ludralph** [reply]
Hi @vipertecpro, you can also use the `default` attribute when creating your schema so you don't have to pass values for `created_at` and `updated_at`. For example:
```prisma
 createdAt      DateTime @default(now())
```
